pkg/controller/dns/external_dns_reconciler_test.go

```diff
@@ -0,0 +1,11 @@
+package dns
+
+var (
+	privateZoneOne = "/subscriptions/test-subscription/resourceGroups/test-rg-private/providers/Microsoft.Network/privatednszones/test-one.com"
```

Comment:
there's no tests actually here?

Reply:
this was meant to include tests for DNS Zone config generation, but moving most of that logic to config anyway so will put testing there - thanks for catching that!

Reply:
added testing and moved logic in next commit, marking as resolved